File: 刷题/cpp/1042. 不邻接植花.cpp
class Solution {
public:
    vector<int> gardenNoAdj(int n, vector<vector<int>>& paths) {
        vector<vector<int> > g(n);
        for(auto path:paths){
            int x=path[0]-1,y=path[1]-1;
            g[x].push_back(y);
            g[y].push_back(x);
        }
        vector<int> answer(n);
        for(int i=0;i<n;i++){
            vector<bool> colored(5, false);
            for(auto &v:g[i]){
                colored[answer[v]]=true;
            }
            for(int j=1;j<=4;j++){
                if(!colored[j]){
                    answer[i]=j;
                    break;
                }
            }
        }
        return answer;
    }
};

File: 刷题/golang/1042. 不邻接植花.go
func gardenNoAdj(n int, paths [][]int) []int {
	ans := make([]int, n)
	adj := make([][]int, n)
	for i := 0; i < n; i++ {
		adj[i] = []int{}
	}
	for _, path := range paths {
		x := path[0] - 1
		y := path[1] - 1
		adj[x] = append(adj[x], y)
		adj[y] = append(adj[y], x)
	}
	for i := 0; i < n; i++ {
		colored := make([]bool, 5)
		for _, v := range adj[i] {
			colored[ans[v]] = true
		}
		for j := 1; j <= 4; j++ {
			if !colored[j] {
				ans[i] = j
				break
			}
		}
	}
	return ans
}